IndexationEngine.py

```diff
@@ -0,0 +1,73 @@
+"""
+    Search Engine
+    by Vincent Jeanselme
+    and Matthieu Clin
+    vincent.jeanselme@gmail.com
+
+    Indexation part from the Engine. Here we are indexing all documents in order to make them available trough
+    later search
+"""
+
+import argparse
+import os
+import IndexationFactory
+
+
+class CheckValidPath(argparse.Action):
+    """
+        Checks if each document provided is a valid path for a file
```

Comment:
I tried to implement a custom [action for the argparse] (https://docs.python.org/3/library/argparse.html#action) in order to keep only the valid paths.
At the beginning, I thought that it would throw an error if there was no valid paths (same kind of error you receive when you provide 0 path) but the checking seems to be before the call of this method (which simply set the value into the desired variable.

If you want, I can remove it and make the filtering after the parsing phase.